src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -592,7 +608,7 @@ private SignatureProvider CreateSignatureProvider(SecurityKey key, string algori
                         signatureProvider = new SymmetricSignatureProvider(key, algorithm, willCreateSignatures);
 
                     if (ShouldCacheSignatureProvider(signatureProvider))
-                        CryptoProviderCache.TryAdd(signatureProvider);
+                        signatureProvider.IsCached = CryptoProviderCache.TryAdd(signatureProvider);
```

Comment:
When is this state changed? Is it needed just to immediately dispose signature providers that are not cached?

Reply:
TryAdd could fail.